core/src/main/java/tachyon/master/BlockInfo.java

```diff
@@ -175,21 +162,52 @@ public synchronized InodeFile getInodeFile() {
           } catch (UnknownHostException e) {
             resolvedHost = loc;
           }
-          ret.add(new NetAddress(resolvedHost, -1, -1));
+          ret.add(resolvedHost);
         }
       }
     }
     return ret;
   }
 
+  public synchronized List<NetAddress> getWorkerAddresses() {
+    List<NetAddress> ret = new ArrayList<NetAddress>();
+    ret.addAll(mLocations.values());
+    return ret;
+  }
+
+  /**
+   * Get the list of pairs "blockId, workerId", where the blockId is the id of this block, and the
+   * workerId is the id of the worker who has the block's data in memory.
+   *
+   * @return the list of those pairs
+   */
+  public synchronized List<Pair<Long, Long>> getBlockIdWorkerIdPairs() {
+    List<Pair<Long, Long>> ret = new ArrayList<Pair<Long, Long>>(mLocations.size());
+    for (long workerId : mLocations.keySet()) {
+      ret.add(new Pair<Long, Long>(mBlockId, workerId));
+    }
+    return ret;
+  }
+
+  /**
+   * Get the InodeFile of the block
+   *
+   * @return the InodeFile of the block
+   */
+  public synchronized InodeFile getInodeFile() {
+    return mInodeFile;
+  }
+
   /**
    * @return true if the block is in some worker's memory, false otherwise
    */
   public synchronized boolean isInMemory() {
-    for (long storageDirId : mStorageDirIds.values()) {
-      int storageLevelValue = StorageDirId.getStorageLevelAliasValue(storageDirId);
-      if (storageLevelValue == StorageLevelAlias.MEM.getValue()) {
-        return true;
+    for (Set<Long> storageDirIdSet : mWorkerDirs.values()) {
+      for (long storageDirId : storageDirIdSet) {
```

Comment:
I think if we're storing the actual storage directory ids, then checking for the MEM value (which is going to be 1, right?) wouldn't work.